src/scrape_up/steam/steamScraper.py

```diff
@@ -0,0 +1,146 @@
+        percentage = match.group(2) if match else 'N/A'
+        review_count = match.group(3) if match else 'N/A'
+        review_count = review_count.strip()
+        reviews = f"{sentiment.strip()} - {percentage.strip()}"
+
+        return name, published_date, platforms, original_price, discount_pct, discount_price, reviews, review_count
+
+    def _scrape_page(self, url, filter, n0Games):
+        """
+        Scrapes game data for a given URL and filter.
+
+        Args:
+            url (str): The URL to scrape.
+            filter (str): The filter for the game data.
+
+        Returns:
+            list: List of game data for the filter.
+        """
+        total_pages = self._get_total_pages(url)
+        all_game_info = []
+
+        for page in range(1, total_pages + 1):
+            response = requests.get(f"{url}&page={page}")
+            doc = BeautifulSoup(response.content, 'html.parser')
+            games = doc.find_all('div', {'class': 'responsive_search_name_combined'})
+
+            for game in games:
+                game_info = self._extract_game_info(game)
+                all_game_info.append([*game_info, filter])
+
+                if len(all_game_info) >= n0Games:
+                    break
+            if len(all_game_info) >= n0Games:
+                break
+        return all_game_info
+
+    def Scraper(self, n0Games=5, tags=['Discounts', 'F2P'], csv=0):
+        """
+        Scrapes game data for each tag specified in the 'tags' attribute and returns a DataFrame.
+
+        Returns:
+            DataFrame: DataFrame containing game data.
+
+        Tags:
+            The 'tags' attribute specifies the filters to be applied during scraping.
+            These tags correspond to keys in the 'search_filters' dictionary, which maps each tag to its respective filter query parameter.
+
+            Example:
+            If 'tags' is ['Discounts', 'F2P'], the scraper will retrieve games that are discounted and free to play.
+            The available tags are:
+                Top Sellers, Discounts, F2P (Free to Play)
+                Software, DLC, Demos, Mods, Games
+                Windows, MacOS, Linux, VR, Deck
+                MP, PvP, Co-op
+        """
+        all_data = []
+        search_filters = {
+            'Top Sellers': 'topselling', 'Discounts': 'specials=1', 'F2P': '?maxprice=free', 'Software': 'category1=994',
+            'DLC': 'category1=21', 'Demos': 'category1=10', 'Mods': 'category1=997', 'Games': 'category1=998',
+            'Windows': 'os=win', 'MacOS': 'os=mac', 'Linux': 'os=linux', 'VR': 'vrsupport=402',
+            'Deck': 'deck_compatibility=2', 'MP': 'category3=1', 'PvP': 'category3=49', 'Co-op': 'category3=9'
+        }
+        for filter in tags:
+            url = f'{self.base_url}{search_filters[filter]}'
+            filter_data = self._scrape_page(url, filter, n0Games)
+            all_data.extend(filter_data)
+        df = pd.DataFrame(all_data, columns=self.cols)
```

Comment:
Do not use packages like Pandas, we do not want to make the package bulky including packages like `Pandas`, also there is no requirement of processing CSV, just scrape the data and serve the response.